dataland-frontend/tests/e2e/fixtures/lksg/LksgDataFixtures.ts

```diff
@@ -7,534 +7,371 @@ import {
-  return faker.helpers.arrayElement(Object.values(NationalOrInternationalMarket));
-}
-
-/**
- * Generates an LKSG dataset with the value null for some categories, subcategories and field values.
- * Datasets that were uploaded via the Dataland API can look like this in production.
- * @returns the dataset
- */
-export function generateOneLksgDatasetWithManyNulls(): LksgData {
-  return {
-    general: {
-      masterData: {
-        dataDate: "1999-12-24",
-        headOfficeInGermany: null!,
-        groupOfCompanies: null!,
-        groupOfCompaniesName: null!,
-        industry: null!,
-        numberOfEmployees: null!,
-        seasonalOrMigrantWorkers: null!,
-        shareOfTemporaryWorkers: null!,
-        totalRevenueCurrency: null!,
-        annualTotalRevenue: null!,
-        fixedAndWorkingCapital: null!,
-      },
-      productionSpecific: null!,
-      productionSpecificOwnOperations: null!,
-    },
-    governance: null!,
-    social: null!,
-    environmental: null!,
-  };
-}
-
 /**
  * Generates a random LKSG dataset
- * @param undefinedProbability the ratio of fields to be undefined (number between 0 and 1)
+ * @param undefinedProbability the probability (as number between 0 and 1) for "undefined" values in nullable fields
  * @returns a random LKSG dataset
  */
-export function generateLksgData(undefinedProbability = 0.5): LksgData {
+export function generateLksgData(undefinedProbability = DEFAULT_PROBABILITY): LksgData {
+  const dataGenerator = new LksgGenerator(undefinedProbability);
   return {
     general: {
       masterData: {
-        dataDate: randomFutureDate(),
-        headOfficeInGermany: valueOrUndefined(randomYesNo(), undefinedProbability),
-        groupOfCompanies: valueOrUndefined(randomYesNo(), undefinedProbability),
-        groupOfCompaniesName: valueOrUndefined(faker.company.name(), undefinedProbability),
-        industry: valueOrUndefined(generateListOfRandomNaceCodes(), undefinedProbability),
-        numberOfEmployees: valueOrUndefined(randomInt(10000), undefinedProbability),
-        seasonalOrMigrantWorkers: valueOrUndefined(randomYesNo(), undefinedProbability),
-        shareOfTemporaryWorkers: valueOrUndefined(randomShareOfTemporaryWorkersInterval(), undefinedProbability),
-        annualTotalRevenue: valueOrUndefined(randomFloat(1000000, 10000000000, 1), undefinedProbability),
-        totalRevenueCurrency: valueOrUndefined(generateIso4217CurrencyCode(), undefinedProbability),
-        fixedAndWorkingCapital: valueOrUndefined(randomInt(10000), undefinedProbability),
+        dataDate: generateFutureDate(),
+        headOfficeInGermany: dataGenerator.randomYesNo(),
+        groupOfCompanies: dataGenerator.randomYesNo(),
+        groupOfCompaniesName: dataGenerator.valueOrUndefined(faker.company.name()),
+        industry: dataGenerator.valueOrUndefined(generateListOfRandomNaceCodes()),
+        numberOfEmployees: dataGenerator.valueOrUndefined(generateInt(10000)),
```

Comment:
The dataGenerator already has a `randomInt()` method you could use here.

Reply:
changed